packages/botonic-react/src/components/webchat-settings.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { useContext } from 'react'
+import { renderComponent, serializeRegexs } from '../utils'
+import { WebchatContext } from '../contexts'
+import { INPUT } from '@botonic/core'
+
+export const WebchatSettings = ({
+  theme,
+  blockInputs,
+  persistentMenu,
+  enableEmojiPicker,
+  enableAttachments,
+}) => {
+  const renderBrowser = () => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const { updateWebchatDevSettings } = useContext(WebchatContext)
+    updateWebchatDevSettings({
+      theme,
+      blockInputs,
+      persistentMenu,
+      enableEmojiPicker,
+      enableAttachments,
+    })
+    return null
+  }
+  const renderNode = () => {
+    const updatedTheme = normalizeWebchatSettings({
+      theme,
+      blockInputs,
+      persistentMenu,
+      enableEmojiPicker,
+      enableAttachments,
+    })
+    return (
+      <message
+        type={INPUT.WEBCHAT_SETTINGS}
+        settings={JSON.stringify({ theme: updatedTheme }, serializeRegexs)}
+      ></message>
+    )
+  }
+  return renderComponent({ renderBrowser, renderNode })
+}
+
+export const normalizeWebchatSettings = settings => {
+  const {
+    theme,
+    blockInputs,
+    persistentMenu,
+    enableEmojiPicker,
+    enableAttachments,
+  } = settings
+  if (!theme.userInput) theme.userInput = {}
+  if (persistentMenu !== undefined) {
```

Comment:
did you choose "!== undefined" because you want to activate the "if" when persistentMenu is null?

Reply:
a really good observation! 💯  yes, this is done in order to allow nullish values. Let's say you want to disable the persistentMenu again, so you should pass `persistentMenu={nulll}` prop.